tests/test_taskgroups.py

```diff
@@ -542,6 +542,26 @@ async def cancel_when_ready():
     assert not sleep_completed
 
 
+async def test_shielding_mutate():
+    done = False
+
+    async def task(task_status):
+        nonlocal done
+        with CancelScope() as scope:
+            await sleep(.1)
+            scope.shield = True
+            task_status.started()
```

Comment:
Since cancellation is only attempted after this line, why not create the cancel scope with `shield=True`?

Reply:
My intention is to have coverage for the False -> True transition while inside the scope.  Even though we know this case isn't significant for the implementation, ideally the test shouldn't assume that.

Reply:
My intention is to have coverage for the False -> True transition while inside the scope.  Even though we know this case isn't significant for the implementation, ideally the test shouldn't assume that.